allennlp/tests/training/trainer_test.py

```diff
@@ -42,16 +43,18 @@ def setUp(self):
         )
         self.model = SimpleTagger.from_params(vocab=self.vocab, params=self.model_params)
         self.optimizer = torch.optim.SGD(self.model.parameters(), 0.01, momentum=0.9)
-        self.iterator = BasicIterator(batch_size=2)
-        self.iterator.index_with(vocab)
+        self.data_loader = DataLoader(self.instances, batch_size=2, collate_fn=allennlp_collocate)
```

Comment:
You can either pass a batch size, in which case it will use a uniform random sampler, or you can pass a `Sampler` (which do not take a batch size) with a specified `batch_size` passed here, _or_ you can specify only a `BatchSampler` - which is mutually exclusive with the other two options.